.github/workflows/general.yaml

```diff
@@ -0,0 +1,59 @@
+name: "general"
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  cargo_test:
+    name: cargo_test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: 1.84.0
```

Comment:
These machines have `rustup`. Might be simpler to just provide a `rust-toolchain.toml` and let `rustup` grab the version rather than using an action where we have to keep in sync with what we're using locally